CRM/Case/XMLProcessor/Process.php

```diff
@@ -568,6 +572,106 @@ public function createActivity($activityTypeXML, &$params) {
     return TRUE;
   }
 
+  /**
+   * Return the default assignee contact for the activity.
+   *
+   * @param array $activityParams
+   * @param object $activityTypeXML
+   * @return int|null the ID of the default assignee contact or null if none.
+   */
+  protected function getDefaultAssigneeForActivity($activityParams, $activityTypeXML) {
+    if (!isset($activityTypeXML->default_assignee_type)) {
+      return NULL;
+    }
+
+    $this->setupDefaultAssigneeOptionIds();
+
+    switch($activityTypeXML->default_assignee_type) {
+      case $this->defaultAssigneeOptionsIds['BY_RELATIONSHIP']:
+        return $this->getDefaultAssigneeByRelationship($activityParams, $activityTypeXML);
+        break;
+      case $this->defaultAssigneeOptionsIds['SPECIFIC_CONTACT']:
+        return $this->getDefaultAssigneeBySpecificContact($activityTypeXML);
+        break;
+      case $this->defaultAssigneeOptionsIds['USER_CREATING_THE_CASE']:
+        return $activityParams['source_contact_id'];
+        break;
+      case $this->defaultAssigneeOptionsIds['NONE']:
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Fetches and caches the activity's default assignee options.
+   *
+   * @return array
+   */
+  protected function setupDefaultAssigneeOptionIds() {
+    if (!empty($this->defaultAssigneeOptionsIds)) {
+      return $defaultAssigneeOptionsIds;
+    }
+
+    $defaultAssigneeOptions =  civicrm_api3('OptionValue', 'get', array(
+      'option_group_id' => 'activity_default_assignee',
+      'options' => array('limit' => 0)
+    ));
+
+    foreach($defaultAssigneeOptions['values'] as $option) {
+      $this->defaultAssigneeOptionsIds[$option['name']] = $option['id'];
```

Comment:
Option Values ids can change for many reasons and they are not used as FKs. We use the value instead. For example, consider we have these option values for default assignee:

|id|name|value|
|-|-----|------|
|1|none|2|
|2|by_relationship|3|
|3|specific_contact|4| 

Then we have an activity definition where the default assignee is by_relationship. In that case, what should be stored in the XML is the 3 (the value) and not 2 (the ID).

I didn't check the other PR where the Option Group was added, but in case you're storing the Option Value id there, it needs to be fixed.